src/dynamite/_backend/bsubspace_impl.h

```diff
@@ -299,6 +303,23 @@ static inline PetscInt S2I_Explicit(PetscInt state, const data_Explicit* data) {
   PetscInt left, right, mid;
   left = 0;
   right = data->dim-1;
+  
+  if (data->rmap_indices == NULL) {
+    while (left <= right) {
+      mid = (left + right)/2;
+      if (data->rmap_states[mid] == state) {
+        return mid;
```

Comment:
I think we don't need to duplicate all of this---since we only use rmap_indices in the return statement, we can probably put the switch on rmap_indices being NULL there. Thoughts?